### PR TITLE
[12.0] [MIG] hr_contract_reference

### PR DESCRIPTION
**Pull request by Nikul-Chaudhary**
HR Contract Reference
=====================

This module was written to extend the functionality of employees contracts to support sequence of contract reference which will be generated automatically from the sequence predefined.

### CODE REVIEW

**Comment by Nikul-Chaudhary** [comment]
@pedrobaeza Could you review please it?

File: hr_contract_reference/README.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
```
Comment (tarteo) [inline hr_contract_reference/README.rst]: Use the split readme

**Review by tarteo — CHANGES_REQUESTED** [review]
Can you increase test coverage?

File: hr_contract_reference/README.rst (hunk 1 above)
Comment (Nikul-Chaudhary) [reply hr_contract_reference/README.rst]: > Use the split readme

Done

**Comment by Nikul-Chaudhary** [comment]
@tarteo @pedrobaeza COuld you please review it ?

**Comment by tarteo** [comment]
@Nikul-Chaudhary Increase the test coverage

**Review by tbaden — CHANGES_REQUESTED** [review]
code lgtm
please use the [maintainer-tools](https://github.com/OCA/maintainer-tools) to generate the hr_contract_reference/readme file

**Comment by jarroyomorales** [comment]
@Nikul-Chaudhary Will you keep working on this?

**Comment by Nikul-Chaudhary** [comment]
> @Nikul-Chaudhary Will you keep working on this?

Yes Sure @jarroyomorales

**Comment by Nikul-Chaudhary** [comment]
@tarteo @jarroyomorales Can you please review it?

**Review by jarroyomorales — APPROVED** [review]
Nice! If you could increase a little bit the coverage it would be fine, but LGTM :+1:

**Comment by Nikul-Chaudhary** [comment]
@OCA/human-resources-maintainers Can you please code review?

File: hr_contract_reference/readme/INSTALL.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+To install this module, you need to:
```
Comment (alexey-pelykh) [inline hr_contract_reference/readme/INSTALL.rst]: There details are useless, they are the same for all modules

File: hr_contract_reference/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+When you will create a new employee contract, the field reference will be
+assigned automatically with the next number of the predefined sequence.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
```
Comment (alexey-pelykh) [inline hr_contract_reference/readme/USAGE.rst]: To my understanding, this part is not needed and is auto-added?

**Review by alexey-pelykh — CHANGES_REQUESTED** [review]

**Comment by jarroyomorales** [comment]
@Nikul-Chaudhary Hi! How is this going? :smiley:

**Comment by jarroyomorales** [comment]
Hi @Nikul-Chaudhary, I just did a PR to your branch fixing the readme as suggested by @alexey-pelykh and added a mini test to increase coverage. Can you check?
https://github.com/Nikul-Chaudhary/hr/pull/1

**Comment by jarroyomorales** [comment]
@Nikul-Chaudhary Could you squash the 3 commits?

**Comment by Saran440** [comment]
@Nikul-Chaudhary Functional Test, LGTM :+1: 
Could you squash commit?

**Comment by Nikul-Chaudhary** [comment]
@jarroyomorales @Saran440 Sure, I will

**Comment by Nikul-Chaudhary** [comment]
@jarroyomorales @Saran440  Done

**Review by Saran440 — APPROVED** [review]

**Comment by jarroyomorales** [comment]
@Saran440 @pedrobaeza What happened with this merge?

**Review by marcelsavegnago — APPROVED** [review]

**Comment by Saran440** [comment]
@jarroyomorales Maybe this pr have CR and not approved yet.

**Comment by etobella** [comment]
Yes, but all changes have been attended, so we need to wait for them to recheck?

@alexey-pelykh @tbaden @tarteo

File: hr_contract_reference/models/hr_contract.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+from odoo import api, fields, models
+
+
+class HrContract(models.Model):
+    _inherit = 'hr.contract'
+
+    name = fields.Char(
+        'Contract Reference',
+        required=False,
+        readonly=True,
```
Comment (SerpentCS) [inline hr_contract_reference/models/hr_contract.py:13]: Is it really needed to co-exist req=False and readonly=True?

Comment (jarroyomorales) [reply hr_contract_reference/models/hr_contract.py:13]: IMO name should be required, and even more if you are using this module, where you shouldnt have a contract without reference.

**Comment by jarroyomorales** [comment]
Hi @alexey-pelykh! Could you recheck please? Thank you!

**Review by alexey-pelykh — APPROVED** [review]

File: hr_contract_reference/models/hr_contract.py (hunk 4 above)
Comment (alexey-pelykh) [reply hr_contract_reference/models/hr_contract.py:13]: It should be required=False on DB and required=true on UI

**Review by etobella — APPROVED** [review]
LGTM :+1: I think that all comments have been attended, so probably this could go to merge @OCA/human-resources-maintainers

**Pull request merged**